Source/PINDiskCache.m

```diff
@@ -572,6 +576,8 @@ - (BOOL)removeFileAndExecuteBlocksForKey:(NSString *)key
     
     [self unlock];
     
+    [_observerManager deletedValueForKey:key];
```

Comment:
If you do decide to make observerManager lazily loaded, this won't be safe accessed outside of the lock.